### PR TITLE
GH-97592: Fix crash in C remove_done_callback due to evil code

### PR DESCRIPTION
**Pull request by gvanrossum**
* Issue: gh-97592

### CODE REVIEW

**Comment by gvanrossum** [comment]
@xiaxinmeng Feel free to add your review.

**Review by JelleZijlstra — APPROVED** [review]

File: Lib/test/test_asyncio/test_futures.py [hunk 1]
```diff
@@ -832,6 +832,21 @@ def __eq__(self, other):
...
+        fut.add_done_callback(str)
+
+        for _ in range(63):
+            fut.add_done_callback(id)
+
+        class evil:
+            def __eq__(self, other):
+                fut.remove_done_callback(other)
+
+        fut.remove_done_callback(evil())
```
Comment (CharlieZhao95) [inline Lib/test/test_asyncio/test_futures.py:848]: > Class names should normally use the CapWords convention.

```suggestion
        class Evil:
            def __eq__(self, other):
                fut.remove_done_callback(other)

        fut.remove_done_callback(Evil())
```
```

File: Misc/NEWS.d/next/Library/2022-09-29-23-22-24.gh-issue-97592.tpJg_J.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Avoid a crash in the C version of `asyncio.Future.remove_done_callback()` when an evil argument is passed.
```
Comment (kumaraditya303) [inline Misc/NEWS.d/next/Library/2022-09-29-23-22-24.gh-issue-97592.tpJg_J.rst]:
```suggestion
Avoid a crash in the C version of :meth:`asyncio.Future.remove_done_callback` when an evil argument is passed.
```

**Review by kumaraditya303 — APPROVED** [review]
LGTM

File: Lib/test/test_asyncio/test_futures.py (hunk 1 above)
Comment (CharlieZhao95) [inline Lib/test/test_asyncio/test_futures.py:848]: All callbacks added by `fut.add_done_callback()` will be cleared when calling `fut.remove_done_callback(Evil())`, but the return value will be the number of callbacks added last.

I'm not sure if we need to improve the return value of `remove done callback` further, the current changes are good enough to fix the crash. :)

**Review by CharlieZhao95 — CHANGES_REQUESTED** [review]

**Pull request merged**